### PR TITLE
circleci version after doing performance evaluation

### PR DESCRIPTION
**Pull request by dfeldman**
We're trying to switch SPIRE from TravisCI to CircleCI. I did an evaluation of CircleCI here:
https://docs.google.com/document/d/1Yh5E2i_5apod11JYm1z-nLRO4tTWEGp0SSBI9x6yytc/edit?usp=sharing
And it seems good for our purpose. We'd like to run both simultaneously for a bit to iron out any issues, and then cut over entirely to CircleCI. 

**Pull Request check list**

- [X ] Commit conforms to CONTRIBUTING.md?
- [X ] Proper tests/regressions included?
- [X ] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

**Comment by evan2645** [comment]
@dfeldman thank you very much for this contribution.

I have been working on the branch locally for a little while now - I hope to finish up early next week and will push some updates here. Thanks again!

**Comment by evan2645** [comment]
Ok... I've finally got this wrapped up. Sorry for the delay

The commit I pushed on top should summarize the changes fairly well. I've also included some commentary at the top of the config file on exactly which environment variables must be set for everything to work well... I will set them in Circle-CI before merging this PR.

Big thanks to @dfeldman and @azdagron for laying the groundwork

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+  # Build the docker images
+  build-docker-images:
+    docker:
+      - image: circleci/golang:<< pipeline.parameters.go-version >>
+    resource_class: << pipeline.parameters.resource-class >>
+    steps:
+      - setup_remote_docker: {}
+      - checkout
+      - run: make images
+      - run: docker save spire-server:latest-local spire-agent:latest-local k8s-workload-registrar:latest-local | gzip > images.tar.gz
```
Comment (azdagron) [inline .circleci/config.yml:106]: We're missing the oidc-discovery-provider image here...

**Review by azdagron — COMMENTED** [review]
Looking pretty good! My one comment was not noticed because that image isn't used in the integration tests but it would have failed the publishing steps. Not sure if/how we can get confidence with the publishing stuff before we turn it on :man_shrugging:.

**Comment by evan2645** [comment]
> My one comment was not noticed because that image isn't used in the integration tests but it would have failed the publishing steps. Not sure if/how we can get confidence with the publishing stuff before we turn it on 🤷‍♂️.

Ahh yes thank you!

Yes it's hard to test without another account (which I don't have)

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**